pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -488,6 +471,9 @@ func (c *defaultControllerInstallationControl) cleanOldExtensions(ctx context.Co
 	return nil
 }
 
+// cleanOldResources cleans ControllerInstallation resources which are not found in the given <newResourcesSet>.
+//
+// TODO: Kept for backwards compatibility. Remove after several releases.
```

Comment:
Nice! Should we open a separate issue to keep track? We haven't done that in the past, but I feel that the number of such things is increasing and no longer be handleable without issues?

Reply:
Agree, we can open one.